skyd/cache.go

```diff
@@ -0,0 +1,144 @@
+package skyd
+
+import (
+	"sync"
+
+	"gitlab.com/NebulousLabs/errors"
+)
+
+var (
+	// ErrNoRebuildInProgress is returned when we try to finish a rebuild but
+	// none is in progress.
+	ErrNoRebuildInProgress = errors.New("no rebuild in progress")
+	// ErrRebuildInProgress is returned when we try to start a rebuild and one
+	// is already in progress.
+	ErrRebuildInProgress = errors.New("rebuild in progress")
+
+	// skylinksCache is a singleton cache, initialised on module load.
+	skylinksCache *pinnedSkylinksCache
+)
+
+type (
+	// pinnedSkylinksCache is a simple cache of the renter's directory
+	// information, so we don't need to fetch that for each skylink we
+	// potentially want to pin/unpin.
+	pinnedSkylinksCache struct {
+		rebuildCh chan interface{}
```

Comment:
The more common way to do this is `chan struct{}` and then if you ever need to push to it do `c <- struct{}{}`.

Reply:
Oh, you are right about the `struct{}` thing. Will change.